src/core/segment_reader.rs

```diff
@@ -393,3 +393,57 @@ impl<'a> Iterator for SegmentReaderAliveDocsIterator<'a> {
         result
     }
 }
+
+#[cfg(test)]
+mod test {
+    use core::{Index, SegmentReader};
+    use schema::*;
+
+    #[test]
+    fn test_iterator() {
+        let mut schema_builder = SchemaBuilder::new();
+        schema_builder.add_text_field("name", TEXT);
+        let schema = schema_builder.build();
+        let index = Index::create_in_ram(schema.clone());
+        let name = schema.get_field("name").unwrap();
+
+        {
+            let mut index_writer = index.writer(50_000_000).unwrap();
+            let mut doc1 = Document::default();
+            doc1.add_text(name, "tantivy");
```

Comment:
Alternatively you can use the following syntax
`index_writer.add_document(doc!(name=>"tantivy"));`

You can leave it that way it is fine as is.

Reply:
neat! Changed.